golang/checksum.go

```diff
@@ -0,0 +1,82 @@
+package tchannel
+
+import (
+	"hash"
+	"hash/crc32"
+)
+
+// The type code for support checksums
+type ChecksumType byte
+
+const (
+	ChecksumTypeNone     ChecksumType = 0
+	ChecksumTypeCrc32    ChecksumType = 1
+	ChecksumTypeFarmhash ChecksumType = 2
+)
+
+// Returns the size of the checksum for the given type
+func (t ChecksumType) ChecksumSize() int {
+	switch t {
+	case ChecksumTypeNone:
+		return 0
+	case ChecksumTypeCrc32:
+		return crc32.Size
+	case ChecksumTypeFarmhash:
+		return 4
+	default:
+		return 0
+	}
+}
+
+// Creates a new Checksum of the given type
+func (t ChecksumType) New() Checksum {
+	switch t {
+	case ChecksumTypeNone:
+		return NullChecksum{}
+	case ChecksumTypeCrc32:
+		return NewCrc32Checksum()
+	case ChecksumTypeFarmhash:
+		return nil
```

Comment:
What does this mean for callers? Should there be a TODO about support? Or should we make this a NullChecksum if we're never going to support this?